shared/modules/treestore_module.go

```diff
@@ -21,13 +21,16 @@ type TreeStoreModule interface {
 	Submodule
 	treeStoreFactory
 
-	// Update returns the new state hash for a given height.
+	AtomicStore
+
+	// Update returns the computed state hash for a given height.
 	// * Height is passed through to the Update function and is used to query the TxIndexer for transactions
 	// to update into the merkle tree set
 	// * Passing a higher height will cause a change but repeatedly calling the same or a lower height will
 	// not incur a change.
 	// * By nature of it taking a pgx transaction at runtime, Update inherits the pgx transaction's read view of the
 	// database.
+	// * Commit must be called after Update to persist any changes it made to disk.
 	Update(pgtx pgx.Tx, height uint64) (string, error)
```

Comment:
I have considered renaming this, yes. Might I propose `GetAndUpdate` or `FetchAndUpdate`.

Reply:
I like your idea and was going to suggest `QueryAndUpdate` given that it takes `pgtx` as an argument. However, maybe `CollectAndUpdate` to avoid being implementation specific but capture the fact that it "collects" all the changes. THoughts?